components/layout_2020/flexbox.rs

```diff
@@ -104,18 +118,76 @@ where
                 ),
             )))
         } else {
-            ArcRefCell::new(FlexLevelBox::FlexItem(IndependentFormattingContext::construct(
+            ArcRefCell::new(FlexLevelBox::FlexItem(
+                IndependentFormattingContext::construct(
+                    self.context,
+                    node,
+                    style.clone(),
+                    display_inside,
+                    contents,
+                    ContentSizesRequest::None, // FIXME: request sizes when we start using them
+                    self.text_decoration_line,
+                ),
+            ))
+        };
+        self.children.push(box_.clone());
+        box_slot.set(LayoutBox::FlexLevel(box_))
+    }
+}
+
+/// https://drafts.csswg.org/css-text/#white-space
+fn is_only_document_white_space(string: &str) -> bool {
```

Comment:
is this something that should be cached?

Reply:
It could be, but I’m not sure that would be a win. I’d guess that white-space only text nodes are typically small (at most a couple newlines plus indentation in non-minified HTML) so this should run quickly, and in other cases we’ll typically find a non-white-space character not far from the start where `Iterator::all` will short-circuit and we’ll also return quickly.